### ISSUE TITLE
Bug in F1 score?

### ISSUE THREAD

**Issue by saskra** [talboger/fastdist#9]
This -->
```python
import numpy as np
from fastdist.fastdist import f1_score

y_true = np.random.randint(2, size=100).astype(bool)
y_pred = np.random.randint(2, size=100).astype(bool)
f1 = f1_score(y_true, y_pred)
```

Results in this -->
```shell
Traceback (most recent call last):
  File "/lib/python3.6/site-packages/numba/core/dispatcher.py", line 420, in _compile_for_args
    error_rewrite(e, 'typing')
  File "/lib/python3.6/site-packages/numba/core/dispatcher.py", line 361, in error_rewrite
    raise e.with_traceback(None)
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Failed in nopython mode pipeline (step: nopython frontend)
Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function iadd>) found for signature:
 
 >>> iadd(Literal[int](0), unicode_type)
 
There are 16 candidate implementations:
  - Of which 14 did not match due to:
  Overload of function 'iadd': File: <numerous>: Line N/A.
    With argument(s): '(int64, unicode_type)':
   No match.
  - Of which 2 did not match due to:
  Operator Overload in function 'iadd': File: unknown: Line unknown.
    With argument(s): '(int64, unicode_type)':
   No match for registered cases:
    * (int64, int64) -> int64
    * (int64, uint64) -> int64
    * (uint64, int64) -> int64
    * (uint64, uint64) -> uint64
    * (float32, float32) -> float32
    * (float64, float64) -> float64
    * (complex64, complex64) -> complex64
    * (complex128, complex128) -> complex128
During: typing of intrinsic-call at /lib/python3.6/site-packages/fastdist/fastdist.py (1281)
File "../../../anaconda3/envs/lib/python3.6/site-packages/fastdist/fastdist.py", line 1281:
def confusion_matrix(targets, preds, w=None, normalize=None):
    <source elided>
                if targets[val] == i and preds[val] == j:
                    correct += w[val]
                    ^
During: resolving callee type: type(CPUDispatcher(<function confusion_matrix at 0x7f046c83b510>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1460)
During: resolving callee type: type(CPUDispatcher(<function confusion_matrix at 0x7f046c83b510>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1505)
During: resolving callee type: type(CPUDispatcher(<function confusion_matrix at 0x7f046c83b510>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1460)
During: resolving callee type: type(CPUDispatcher(<function confusion_matrix at 0x7f046c83b510>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1505)
During: resolving callee type: type(CPUDispatcher(<function confusion_matrix at 0x7f046c83b510>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1460)
File "../../../anaconda3/envs/lib/python3.6/site-packages/fastdist/fastdist.py", line 1460:
def precision_score(targets, preds, cm=None, w=None, average='binary'):
    <source elided>
        if cm is None:
            cm = confusion_matrix(targets, preds, w=w)
            ^
During: resolving callee type: type(CPUDispatcher(<function precision_score at 0x7f046c846048>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1624)
During: resolving callee type: type(CPUDispatcher(<function precision_score at 0x7f046c846048>))
During: typing of call at /lib/python3.6/site-packages/fastdist/fastdist.py (1624)
File "../../../anaconda3/envs/lib/python3.6/site-packages/fastdist/fastdist.py", line 1624:
def f1_score(targets, preds, w=None, average='binary'):
    <source elided>
    w = init_w(w, len(targets))
    precision = precision_score(targets, preds, w, average)
    ^
python-BaseException
```

btw: You could add the precalculated confusion matrix parameter here as well.

**Comment by talboger** [comment]
Good point about adding the confusion matrix here! That's actually the cause for this issue; when I added the confusion matrix parameter to some of the functions, I did not also change the parameters in the F1 score. Currently, the definition of precision score is:

```python
def precision_score(targets, preds, cm=None, w=None, average='binary'):
```

The current F1 score code was not updated for the cm parameter, so it's trying to read `w` as a cm and `average` as `w` because I didn't use keyword parameters. I'll be sure to update this in a future version. Until then though, you can use the following equivalent F1 score:

```python
@jit(nopython=True, fastmath=True)
def f1_score(targets, preds, w=None, average='binary'):
    w = init_w(w, len(targets))
    precision = precision_score(targets, preds, w=w, average=average)
    recall = recall_score(targets, preds, w=w, average=average)
    return np.array([2]) * precision * recall / (precision + recall)
```

**Comment by talboger** [comment]
Sorry for the delay - this has now been fixed in v1.1.3
